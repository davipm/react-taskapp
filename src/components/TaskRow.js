import React from "react";
import PropTypes from "prop-types";

const TaskRow = ({ task, toggleTask, removeTask }) => (
  <tr key={task.id} data-testid="task-row">
    <td data-testid="task-name">{task.name}</td>
    <td>
      <input
        type="checkbox"
        checked={task.done}
        onChange={toggleTask}
        data-testid="toggle"
      />

      <button
        className="btn btn-danger ml-3"
        onClick={() => removeTask(task.id)}
        data-testid="submit"
      >
        Remove
      </button>
    </td>
  </tr>
);

TaskRow.propTypes = {
  toggleTask: PropTypes.func.isRequired,
  removeTask: PropTypes.func.isRequired,
  task: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    done: PropTypes.bool.isRequired,
  }).isRequired,
};

export default TaskRow;
