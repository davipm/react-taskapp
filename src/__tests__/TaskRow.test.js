import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import TaskRow from "../components/TaskRow";

beforeAll(cleanup);

const data = { name: "Task One", done: false, id: 0 };

describe("test <TaskRow /> component", function () {
  it("should render the task", function () {
    const removeTask = jest.fn();
    const toggleTask = jest.fn();

    const { getByTestId } = render(
      <TaskRow task={data} removeTask={removeTask} toggleTask={toggleTask} />
    );

    expect(getByTestId("task-row")).toBeTruthy();
    expect(getByTestId("task-name").textContent).toBe("Task One");
  });

  it("should show toggle and submit component", function () {
    const removeTask = jest.fn();
    const toggleTask = jest.fn();

    const { getByTestId } = render(
      <TaskRow task={data} removeTask={removeTask} toggleTask={toggleTask} />
    );

    fireEvent.click(getByTestId("toggle"));
    expect(toggleTask).toHaveBeenCalledTimes(1);
    expect(getByTestId("toggle").checked).toEqual(false);

    fireEvent.click(getByTestId("toggle"));
    expect(toggleTask).toHaveBeenCalledTimes(2);

    fireEvent.click(getByTestId("submit"));
    expect(removeTask).toHaveBeenCalledTimes(1);
  });

  it("should take snapshot", function () {
    const removeTask = jest.fn();
    const toggleTask = jest.fn();

    const { container } = render(
      <TaskRow task={data} removeTask={removeTask} toggleTask={toggleTask} />
    );

    expect(container).toMatchSnapshot();
  });
});
