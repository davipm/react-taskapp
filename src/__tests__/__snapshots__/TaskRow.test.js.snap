// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <TaskRow /> component should take snapshot 1`] = `
<div>
  <tr
    data-testid="task-row"
  >
    <td
      data-testid="task-name"
    >
      Task One
    </td>
    <td>
      <input
        data-testid="toggle"
        type="checkbox"
      />
      <button
        class="btn btn-danger ml-3"
        data-testid="submit"
      >
        Remove
      </button>
    </td>
  </tr>
</div>
`;
